code/part-one/validation.js

```diff
@@ -4,11 +4,22 @@ const { createHash } = require('crypto');
 const signing = require('./signing');
 
 const isValidTransaction = transaction => {
+  const toSign = transaction.source
+    + transaction.recipient
+    + transaction.amount;
 
+  return signing.verify(transaction.source, toSign, transaction.signature);
 };
 
 const isValidBlock = block => {
+  const transactionString = block.transactions.map(t => t.signature).join('');
+  const toHash = block.previousHash + transactionString + block.nonce;
 
+  if (block.hash !== createHash('sha512').update(toHash).digest('hex')) {
+    return false;
+  }
+
+  return block.transactions.every(t => isValidTransaction(t));
```

Comment:
I think this can be shortened to `return block.transactions.every(isValidTransaction);`